packages/nimble-components/src/table/components/cell/index.ts

```diff
@@ -14,10 +20,58 @@ declare global {
  * A styled cell that is used within the nimble-table-row.
  * @internal
  */
-export class TableCell extends FoundationElement {
-    // TODO: This should be replaced with an instance of TableCellState<TCellData>
+export class TableCell<
+    TCellRecord extends TableRecord = TableRecord
+> extends FoundationElement {
     @observable
-    public data: TableFieldValue;
+    public data?: TableCellState<TCellRecord>;
```

Comment:
can we call this cellState to reflect the type better and so data only refers to the root table data?